temporalcli/commandsgen/parse.go

```diff
@@ -45,6 +47,16 @@ type (
 		Options                []Option `yaml:"options"`
 		OptionSets             []string `yaml:"option-sets"`
 		Docs                   Docs     `yaml:"docs"`
+		Index                  int
+		Base                   *Command
+		Parent                 *Command
+		Children               []*Command
```

Comment:
A lot of times these can be determined dynamically at runtime instead of pre-storing references. See how the code generator handles subcommands and such. I think it's more valuable keeping this field set to the bare minimum than precalculating this information (no perf concerns). So `FileName` may still make sense, but many other things can be determined during generation (e.g. subcommands) or passed as state during generation (e.g. parent).

Reply:
For me having these pre-calculated made `docs.go` rendering cleaner and more focused on the rendering vs. calculating fields in `docs.go`.

The single pass `docs.go` rendering we have today , doesn't provide much latitude for 2nd-order sorting of subcommands e.g.
<img width="1334" alt="image" src="https://github.com/user-attachments/assets/af9817c4-4e85-4b10-a866-4cf6a474e567">

That said, we can always leave this out for now, and bring it back in later if needed.

Reply:
> For me having these pre-calculated made docs.go rendering cleaner and more focused on the rendering vs. calculating fields in docs.go.

Made rendering cleaner for docs only at a cost of expanded/unnecessary/easily-derived state on the model objects. The reason I suggested bringing the "enrich" in the main part is because it looked like it was general purpose things that were missing.

Obviously this isn't a big deal, but ideally the docs code like the code-gen code can derive what it needs when it needs. It'd be nicer to keep the model small, but I won't hold up review.

Reply:
> though will still wait on removing the fields that are never used

Removed the unused fields.

> ideally the docs-gen code like the code-gen code can derive what it needs when it needs.

just saw that [`code.go` is scanning for subcommands for each command](https://github.com/temporalio/cli/pull/691#discussion_r1823258517). Can make it consistent if that's the approach we're using in this package, esp. since performance isn't an issue.

Reply:
Moved to derived state.